packages/server/modules/workspaces/graph/resolvers/workspaces.ts

```diff
@@ -784,6 +791,28 @@ export = FF_WORKSPACES_MODULE_ENABLED
               db
             })
           })({ userId: ctx.userId!, workspaceId: args.input.workspaceId })
+        },
+        requestToJoin: async (_parent, args, ctx) => {
+          const transaction = await db.transaction()
+          return await withTransaction(
+            requestToJoinWorkspaceFactory({
+              createWorkspaceJoinRequest: createWorkspaceJoinRequestFactory({ db }),
+              sendWorkspaceJoinRequestReceivedEmail:
+                sendWorkspaceJoinRequestReceivedEmailFactory({
+                  renderEmail,
+                  sendEmail,
+                  getServerInfo,
+                  getWorkspaceCollaborators: getWorkspaceCollaboratorsFactory({ db }),
+                  getUserEmails: findEmailsByUserIdFactory({ db })
+                }),
+              getUserById: getUserFactory({ db }),
+              getWorkspace: getWorkspaceFactory({ db })
+            })({
```

Comment:
Doing withTransaction + build factory + invoke the factory all in one call is pretty confusing to read. I would at least store the built factory in a const first